src/widgets/canvas.rs

```diff
@@ -176,6 +227,106 @@ impl Grid for CharGrid {
     }
 }
 
+/// The HalfBlockGrid is a grid made up of cells each containing a half block character.
+/// This makes it possible to draw shapes with a resolution of 1x2 dots per cell. This is useful
+/// when you want to draw shapes with a higher resolution than a CharGrid but lower than a
+/// BrailleGrid. This grid type supports a foreground and background color for each terminal cell.
+/// This allows for more flexibility than the BrailleGrid which only supports a single foreground
+/// color for each 2x4 dots cell.
+#[derive(Debug, Default, Clone, Eq, PartialEq, Hash)]
+struct HalfBlockGrid {
+    /// width of the grid in number of terminal columns
+    width: u16,
+    /// height of the grid in number of terminal rows
+    height: u16,
+    /// represents a single color for each cell
+    cells: Vec<Vec<Color>>,
```

Comment:
Above in the comment, there's this text:

> /// ... This grid type supports a foreground and background color for each terminal cell.

I think the name of `cells` field is not matching the usage of `cells` in the comment.

`Cell` refers to a single "pixel" or single "character" in the terminal, but this `Vec<Vec<Color>>` is the color of each half block, right? Maybe this field name should be called `blocks`?

Reply:
pixels sounds right

Reply:
Thinking about it some more, the name `half_blocks` would be clearest imo. But pixels sounds good too!

Reply:
I find calling it `half block` confusing as the half block character takes up a full cell. It's a weird ambiguous semantic thing where I think pixels makes more sense.

Reply:
https://github.com/ratatui-org/ratatui/pull/550/commits/30fbec7f5c91f59e5a0a20d8eccb4a3bae5c916b